acc_acc_modules/1_annotate_acc_modules.py

```diff
@@ -0,0 +1,373 @@
+    a given list of experiment ids (found in the metadata)
+
+    Arguments
+    ----------
+    metadata_filename: str
+        File containing metadata
+    experiment_colname: str
+        Column header that contains experiment id that maps expression data
+        and metadata
+    sample_colname: str
+        Column header that contains sample id that maps expression data
+        and metadata
+    experiment_id: str
+        Selected experiment id to grab samples from
+
+    """
+    # Read in metadata
+    metadata = pd.read_csv(metadata_filename, header=0)
+    metadata.set_index(experiment_colname, inplace=True)
+
+    selected_metadata = metadata.loc[experiment_id]
+    sample_ids = list(selected_metadata[sample_colname])
+
+    return sample_ids
+
+
+# +
+# TO DO: Have Deb or Georgia select a study
+# Looks like we removed many of the clinical isolates from this compendium with our strain binning
+# selected_sample_ids = get_sample_ids(
+#   metadata_filename, experiment_colname="SRA_study", sample_colname="Experiment", experiment_id="SRP063289")
+
+# +
+# Subset compendium
+# subset_pao1_compendium = pao1_compendium.loc[selected_sample_ids]
+# subset_pa14_compendium = pa14_compendium.loc[selected_sample_ids]
+
+# +
+# print(subset_pao1_compendium.shape)
+# print(subset_pa14_compendium.shape)
+
+# +
+# pao1_median_subset = subset_pao1_compendium.median().to_frame("median subset expression")
+# pa14_median_subset = subset_pa14_compendium.median().to_frame("median subset expression")
+# -
+
+# Add median expression to gene ids
+pao1_gene_annot = pao1_gene_module_labels.merge(
+    pao1_median_all, left_index=True, right_index=True, how="left"
+)
+pa14_gene_annot = pa14_gene_module_labels.merge(
+    pa14_median_all, left_index=True, right_index=True, how="left"
+)
+
+# Add median subset expression to gene ids
+"""pao1_gene_annot = pao1_gene_annot.merge(
+    pao1_median_subset, left_index=True, right_index=True, how="left"
+)
+pa14_gene_annot = pa14_gene_annot.merge(
+    pa14_median_subset, left_index=True, right_index=True, how="left"
+)"""
+
+print(pao1_gene_annot.shape)
+pao1_gene_annot.head()
+
+print(pa14_gene_annot.shape)
+pa14_gene_annot.head()
+
+
+# ## Genome location information
+#
+# How far are genes from other genes in the same module?
+
+# +
+# TO DO: Move into scripts
+# For genes in the same module, calculate the pairwise distance from each other
+# Calculate the median pairwise distance to represent how spread the module is
+# across the genome
+# Other metrics?
+
+
+def get_intra_module_dist(annot_df, pa_prefix):
+    rows = []
+    for grp_name, grp_df in annot_df.groupby("module id"):
+        # print("module", grp_name)
+
+        # Trim off "PA" and convert number to integer
+        ids = grp_df.index
+
+        # Convert trailing id numbers to floats
+        num_ids = [float(_id.split(pa_prefix)[1]) for _id in ids]
+
+        abs_dist = []
+        for gene1, gene2 in product(num_ids, num_ids):
```

Comment:
This is probably just ignorance on my part, but what does `product()` do here? I'm not sure how you're getting genomic locations out of this.

Reply:
So `num_ids` is a list of the the gene ids, which we're using as our genomic location. Say our list is ["PA0001", "PA0004", "PA0010"], then product will return an iterator where at the first iteration gene1 = "PA0001" and gene2="PA0001" so there is a distance of 0 here. But in the next iteration we have gene1 = "PA0001" and gene2 = "PA0004"